MetaMask/metamask-extension
# Feat/18890/button ts update

## Explanation
- Update `Button` unification component to TS 
- Update all objects to appropriate enums (e.g. BUTTON_VARIANTS & BUTTON_SIZES to ButtonVariant, ButtonSize)
- Small fixes to app components built in TS using `Button` and created a few lint errors (see comments) 

* Fixes #18890 




## Screenshots/Screencaps



### Before


<img width="858" alt="Screenshot 2023-08-21 at 3 04 15 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/cc822a2d-d8ee-4fd4-a115-30e18e9dd945">
<img width="744" alt="Screenshot 2023-08-21 at 3 03 52 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/6bdfd3d4-fbb1-43e6-a413-87f16966cd51">
<img width="1067" alt="Screenshot 2023-08-21 at 3 03 15 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/d99c3604-0e28-4066-a97c-8059fd0b635f">


### After


<img width="825" alt="Screenshot 2023-08-21 at 3 04 10 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/e7446f7b-4891-48b2-a864-e33afecef395">
<img width="665" alt="Screenshot 2023-08-21 at 3 03 36 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/80943dcd-982c-4b96-bd5f-9356f8050138">
<img width="1029" alt="Screenshot 2023-08-21 at 3 02 34 PM" src="https://github.com/MetaMask/metamask-extension/assets/26469696/35b9d778-5f71-43c4-9bf1-5c148613a03f">



## Manual Testing Steps
`yarn jest ui/components/component-library`

To ensure TS validation is working for size try to use the wrong size value on a `ButtonVariant.Primary` or `ButtonVariant.Secondary` 

Example code that should result in TS error: 
```tsx
<Button variant={ButtonVariant.Primary} size={ButtonSize.Auto}>Button Primary</Button>
```

Video example: https://www.loom.com/share/aa28283f8f5d40b8ad322f38bd7acb36




## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
